src/_zkapauthorizer/recover.py

```diff
@@ -187,11 +177,19 @@ async def canned_downloader(set_state: SetState) -> Awaitable[BinaryIO]:
 noop_downloader = make_canned_downloader(b"")
 
 
-def statements_from_download(data: BinaryIO) -> Iterator[str]:
+def statements_from_snapshot(data: BinaryIO) -> Iterator[str]:
     """
     Read the SQL statements which constitute the replica from a byte string.
     """
-    return data.read().decode("ascii").splitlines()
+    s = data.read()
+    pos = 0
+    while pos < len(s):
+        delim = s.index(b":", pos)
```

Comment:
Not asking for changes .. but this doesn't match by understanding of "netstrings" however it _does_ match what Wikipedia says. (FWIW, my understanding is that you'd interpret the first 2 or 4 or whatever bytes as an int and then read that number of bytes as "the string" ... but now I don't know where that understanding came from ;).

Anyway, I think "bytes" here is at least more-clear than unicode.

Reply:
I've usually heard that protocol called "int 16 strings" or "int 32 strings".  Somewhere else I did link to djb's description of the protocol implemented here - https://cr.yp.to/proto/netstrings.txt

Twisted follows the same naming - eg https://docs.twistedmatrix.com/en/stable/api/twisted.protocols.basic.Int16StringReceiver.html and https://docs.twistedmatrix.com/en/stable/api/twisted.protocols.basic.NetstringReceiver.html